### PR TITLE
User and role bug fixed

### PR DESCRIPTION
**Pull request by ajithandrewj**
## Description
The below issue is fixed
1 Incorrect error message, for bad password.

## Type of Change
- [ yes ] Bug fix
- [ no ] New feature
- [ no ] Breaking change
- [ no ] Documentation update

## Checklist
- [ yes ] My code follows the style guidelines of this project
- [ yes ] I have performed a self-review of my own code
- [ yes ] I have commented my code, particularly in hard-to-understand areas
- [ no ] I have made corresponding changes to the documentation
- [ yes ] My changes generate no new warnings
- [ yes ] I have added tests that prove my fix is effective or that my feature works
- [ yes ] New and existing unit tests pass locally with my changes
- [ no ] Any dependent changes have been merged and published in downstream modules
- [ yes ] All the sanity checks have been completed and the sanity test cases have been executed

## Ansible Best Practices
- [ yes ] Tasks are idempotent (can be run multiple times without changing state)
- [ no ] Variables and secrets are handled securely (e.g., using `ansible-vault` or environment variables)
- [ yes ] Playbooks are modular and reusable
- [ no ] Handlers are used for actions that need to run on change

## Documentation
- [ yes ] All options and parameters are documented clearly.
- [ yes ] Examples are provided and tested.
- [ yes ] Notes and limitations are clearly stated.

### CODE REVIEW

File: plugins/modules/user_role_workflow_manager.py [hunk 1]
```diff
@@ -1038,31 +1038,47 @@ def validate_password(self, password, error_messages):
...
+            r'ABC|BCD|CDE|DEF|EFG|FGH|GHI|HIJ|IJK|JKL|KLM|LMN|MNO|NOP|'
+            r'OPQ|PQR|QRS|RST|STU|TUV|UVW|VWX|WXY|XYZ)).*'
+            r'[a-zA-Z0-9!@#$%^&*()_+<>?]{9,20}$'
+        )
 
+        self.log("Checking that the password is 8 to 20 characters long and includes at least three character types.", "DEBUG")
         for password_regex in password_regexs:
             if password_regex.match(password):
-                is_valid_password = True
+                is_valid_password_check1 = True
```
Comment (madhansansel) [inline plugins/modules/user_role_workflow_manager.py:1074]: self.log("Password meets character type and length requirements.", "INFO")

File: plugins/modules/user_role_workflow_manager.py [hunk 2]
```diff
@@ -1038,31 +1038,47 @@ def validate_password(self, password, error_messages):
         Returns:
             None: This function does not return a value, but it may append an error message to `error_messages` if the password is invalid.
         Criteria:
-            - The password must be 8 to 20 characters long.
+            - The password must be 9 to 20 characters long.
             - The password must include characters from at least three of the following classes:
               lowercase letters, uppercase letters, digits, and special characters.
         """
-        is_valid_password = False
+        is_valid_password_check1 = False
```
Comment (madhansansel) [inline plugins/modules/user_role_workflow_manager.py:1045]: meets_character_requirements = False

File: plugins/modules/user_role_workflow_manager.py [hunk 3]
```diff
@@ -1038,31 +1038,47 @@ def validate_password(self, password, error_messages):
...
+        is_valid_password_check2 = False
```
Comment (madhansansel) [inline plugins/modules/user_role_workflow_manager.py:1046]: passes_sequence_repetition_check = False

File: plugins/modules/user_role_workflow_manager.py [hunk 4]
```diff
@@ -1038,31 +1038,47 @@ def validate_password(self, password, error_messages):
...
+            r'OPQ|PQR|QRS|RST|STU|TUV|UVW|VWX|WXY|XYZ)).*'
+            r'[a-zA-Z0-9!@#$%^&*()_+<>?]{9,20}$'
+        )
 
+        self.log("Checking that the password is 8 to 20 characters long and includes at least three character types.", "DEBUG")
         for password_regex in password_regexs:
             if password_regex.match(password):
-                is_valid_password = True
+                is_valid_password_check1 = True
                 break
```
Comment (madhansansel) [inline plugins/modules/user_role_workflow_manager.py:1076]: if not meets_character_requirements:
        self.log("Password failed character type and length validation.", "ERROR")

File: plugins/modules/user_role_workflow_manager.py [hunk 5]
```diff
@@ -1038,31 +1038,47 @@ def validate_password(self, password, error_messages):
...
 
-        if not is_valid_password:
+        self.log("Checking that the password does not contain repetitive or sequential characters.", "DEBUG")
+        if re.match(password_sequence_repetitive_regex, password):
+            is_valid_password_check2 = True
```
Comment (madhansansel) [inline plugins/modules/user_role_workflow_manager.py:1079]: passes_sequence_repetition_check = True
        self.log("Password passed repetitive and sequential character checks.", "INFO")
    else:
        self.log("Password failed repetitive or sequential character validation.", "ERROR")

File: plugins/modules/user_role_workflow_manager.py (hunk 1 above)
Comment (ajithandrewj) [reply plugins/modules/user_role_workflow_manager.py:1074]: Updated the code accordingly.

File: plugins/modules/user_role_workflow_manager.py (hunk 2 above)
Comment (ajithandrewj) [reply plugins/modules/user_role_workflow_manager.py:1045]: Updated the code accordingly.

File: plugins/modules/user_role_workflow_manager.py (hunk 3 above)
Comment (ajithandrewj) [reply plugins/modules/user_role_workflow_manager.py:1046]: Updated the code accordingly.

File: plugins/modules/user_role_workflow_manager.py (hunk 4 above)
Comment (ajithandrewj) [reply plugins/modules/user_role_workflow_manager.py:1076]: Updated the code accordingly.

File: plugins/modules/user_role_workflow_manager.py (hunk 5 above)
Comment (ajithandrewj) [reply plugins/modules/user_role_workflow_manager.py]: Updated the code accordingly.

Comment (ajithandrewj) [reply plugins/modules/user_role_workflow_manager.py]: Also tested the changes Madhan

**Review by madhansansel — APPROVED** [review]

**Pull request merged**